File: Clase 5 - Estructuras Condicionales/Operadores Comparativos.py
# Operadores Comparativos

# print('1' - 1) # error debido al tipado fuerte


# ==: Igualdad
print(1 == 1) # true
print(1 == '1') # false
print(1 == 2) # false
contraseñaCorrecta = "secreto"
contraseñaIngresada = "secreto"
print(contraseñaCorrecta == contraseñaIngresada) # true

# !=: Desigualdad o Inequalidad
print(1 != 1) # false
print(1 != '1') # true
print(1 != 2) # true
print(contraseñaCorrecta != contraseñaIngresada) # false

# >: Mayor que
print(1 > 1) # false
# print(1 > '1') # false # error debido al tipado fuerte
print(1 > 2) # false
print(2 > 1) # true
nivelAgua = 30
nivelCritico = 20
print(nivelAgua > nivelCritico) # true
print("manzana" > "pera") # false
print("manzana" > "mango") # true

# >=: Mayor o igual que
print(1 >= 1) # true
# print(1 >= '1') # false # error debido al tipado fuerte
print(1 >= 2) # false
print(2 >= 1) # true
nivelAgua = 30
nivelCritico = 20
print(nivelAgua >= nivelCritico) # true
print("manzana" >= "pera") # false
print("manzana" >= "mango") # true

# <=: Menor o igual que
print(1 <= 1) # true
# print(1 <= '1') # false # error debido al tipado fuerte
print(1 <= 2) # true
print(2 <= 1) # false
print("manzana" <= "pera") # true

File: Clase 5 - Estructuras Condicionales/Operadores Comparativos.js
// Operadores Comparativos

console.log('1' - 1) // 0 - js tiene tipado debil

// ==: Igualdad
console.log(1 === 1) // true
console.log(1 == '1') // true
console.log(1 === 2) // false
const contraseñaCorrecta = 'secreto'
const contraseñaIngresada = 'secreto'
console.log(contraseñaCorrecta == contraseñaIngresada) // true

// ===: Igualdad con tipado
console.log(1 === 1) // true
console.log(1 === '1') // false
console.log(1 === 2) // false
console.log(contraseñaCorrecta === contraseñaIngresada) // true

// !=: Desigualdad o Inequalidad
console.log(1 !== 1) // false
console.log(1 != '1') // false
console.log(1 !== 2) // true
console.log(contraseñaCorrecta != contraseñaIngresada) // false

// !==: No igualdad con tipado o desigualdad estricta
console.log(1 !== 1) // false
console.log(1 !== '1') // true
console.log(1 !== 2) // true
console.log(contraseñaCorrecta !== contraseñaIngresada) // false

// >: Mayor que
console.log(1 > 1) // false
console.log(1 > '1') // false
console.log(1 > 2) // false
console.log(2 > 1) // true
const nivelAgua = 30
const nivelCritico = 20
console.log(nivelAgua > nivelCritico) // true
console.log('manzana' > 'pera') // false
console.log('manzana' > 'mango') // true

// >=: Mayor o igual que
console.log(1 >= 1) // true
console.log(1 >= '1') // true
console.log(1 >= 2) // false
console.log(2 >= 1) // true
console.log('manzana' >= 'pera') // false
console.log('manzana' >= 'mango') // true

// <=: Menor o igual que
console.log(1 <= 1) // true
console.log(1 <= '1') // true
console.log(1 <= 2) // true
console.log(2 <= 1) // false
console.log('manzana' <= 'pera') // true
